mbuild/compound.py

```diff
@@ -225,8 +247,171 @@ def particles_by_name(self, name):
             if particle.name == name:
                 yield particle
 
+    @property
+    def rigid(self):
+        """Return the 'rigid' status of a Compound.
+
+        rigid() will return True if the Compound contains any particle with
+        a rigid_id != None or the rigid_id of self is != None.
+
+        Returns
+        -------
+        bool
+            True if the Compound or any successor has a rigid_id != None
+
+        """
+        if (any(successor.rigid_id is not None for successor in self.successors()) or            self.rigid_id is not None):
+            return True
+        else:
+            return False
+
+    def max_rigid(self):
+        """Return the maximum rigid body ID contained in the Compound.
+
+        This is usually used by compound.root to determine the maximum
+        rigid_id in the system as a whole.
+
+        Returns
+        -------
+        int or None
+            The maximum rigid body ID contained in the Compound. If no
+            rigid body IDs are found, None is returned
+
+        """
+        if self.rigid:
+            if self.children:
+                return max(successor.rigid_id for successor in self.successors() if successor.rigid_id is not None)
+            else:
+                return self.rigid_id
+        else:
+            return None
+
+    def rigid_ids(self):
+        """Generate rigid_ids for all Particles in the Compound.
+
+        Yields
+        ------
+        int
+            The rigid_id of the next Particle in the Compound
+
+        """
+        for particle in self.particles():
+            yield particle.rigid_id
+
+    def rigid_particles(self, rigid_id=None):
+        """Generate all particles in rigid bodies.
+
+        Parameters
+        ----------
+        rigid_id : int, optional
+            Include only particles with this rigid body ID
+
+        """
+        for particle in self.particles():
+            if particle.rigid_id is not None:
+                yield particle
+
+    def set_rigid(self, name=False, rigid_id=None):
+        """Treat the Compound and all successors as a single rigid body.
+
+        Provides a unique rigid body ID to a Compound and its successors.
+        If 'name' is specified, only sucessors with this name are provided
+        with a rigid_id.  This is useful for Compounds where only some of
+        the particles should be rigid (such as a tethered nanoparticle with
+        a rigid core).
```

Comment:
Can you just add an example as a doctest? See [here](https://github.com/mosdef-hub/mbuild/blob/master/mbuild/lattice.py#L61-L72) for an example